### ISSUE TITLE
How to use precompiled headers

### ISSUE THREAD

**Issue by dvzubarev** [Andersbakken/rtags#900]
Hi,
Does it make any sense to use precompiled headers with rtags?
If so, it seems they do not work at the moment.
Steps to reproduce:
1. start rdm `rdm -vvvvv  --error-limit=0 &> rtags.log &`
2. clone repo https://github.com/dvzubarev/rtags_precompiled_issue.git
3. generate pch `clang++ -x c++-header build/inc/1.hxx -o build/inc/1.hxx.pch`
4. I put in the Makefile -include option `-include "build/inc/1.hxx"`, so run make 
`make -nBk | rc -vvvv -c -`
5. See in rtags.log that -include was stripped 
`Parsed Source(s) successfully: List<Source>(Source(/usr/bin/clang-3.9 -ferror-limit=0 -Wall -fspell-checking -Wno-unknown-warning-option -DRTAGS -isystem /usr/lib/llvm-3.9/lib/clang/3.9.1/include/`
But I expected that clang made use of '-include' option and precompiled headers.

I tried to add `--pch-enabled` option when running rdm. After running make there was an error in rtags.log
`fatal error: '/home/user/.rtags/_home_user_rtags__precompiled__issue_/1085/pch.h' file not found`
 My -include was stripped as earlier.
Anyway this option is supposed to enable automatically generated pch, isn't it?

**Comment by Andersbakken** [comment]
This option is actually meant to make use of existing precompiled headers
but I'm not that surprised that it doesn't work. I never got around to
finishing the support. I believe it's meant to work with the cmake module
cotire which helps with setting up pch. At the core of the problem is that
rtags needs to build the pch file itself since pch files aren't compatible
between versions of clang and, just as importantly, we need to index those
symbols.

In other words. The command to generate the pch file needs to be known by
cmake too. It doesn't currently appear to be working however. I'll see if I
can take a look at it.

Anders

**Comment by cslux** [comment]
Related to #1048
